projects/core/src/lib/components/toolbar/geolocation/geolocation.component.ts

```diff
@@ -0,0 +1,110 @@
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, OnDestroy, OnInit } from '@angular/core';
+import { MapService, WKTHelper } from '@tailormap-viewer/map';
+import { Subject, takeUntil, take } from 'rxjs';
+import { Store } from '@ngrx/store';
+import { FeatureModel } from '@tailormap-viewer/api';
+
+@Component({
+  selector: 'tm-geolocation',
+  templateUrl: './geolocation.component.html',
+  styleUrls: ['./geolocation.component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class GeolocationComponent implements OnInit, OnDestroy {
+
+  private destroyed = new Subject();
+  private featureGeom = new Subject<FeatureModel[]>();
+
+  private activeWatch: number | undefined = undefined;
+
+  public hasGeolocation = navigator.geolocation !== undefined;
+  public isWatching = false;
+  public hasFix = false;
+
+  constructor(
+    private store$: Store,
+    private mapService: MapService,
+    private cdr: ChangeDetectorRef,
+  ) { }
+
+  public ngOnInit(): void {
+    this.mapService.renderFeatures$('geolocation-layer', this.featureGeom.asObservable(), f => {
+      if (f.__fid === 'geolocation-point') {
+        return {
+          styleKey: 'geolocation-point-style',
+          zIndex: 1000,
+          pointType: 'circle',
+          pointSize: 7,
+          pointFillColor: '#6236ff',
+          pointStrokeColor: '#ffffff',
+          pointStrokeWidth: 2,
+        };
+      } else {
+        return {
+          styleKey: 'geolocation-shadow-style',
+          zIndex: 999,
+          fillColor: '#6236ff',
+          fillOpacity: 20,
+          strokeColor: '#6236ff',
+          strokeOpacity: 40,
+        };
+      }
+     }).pipe(takeUntil(this.destroyed)).subscribe();
+  }
+
+  public ngOnDestroy(): void {
+    this.destroyed.next(null);
+    this.destroyed.complete();
+  }
+
+  private positionSuccess(pos: GeolocationPosition) {
+    this.mapService.getProjectionCode$()
+      .pipe(take(1))
+      .subscribe(projectionCode => {
+          const point = WKTHelper.transformWKT(`POINT(${pos.coords.longitude} ${pos.coords.latitude})`, 'EPSG:4326', projectionCode);
```

Comment:
It feels a bit strange to create WKT, then transform that to WKT in a different projection. Maybe its better to add a method to `projection.helper.ts` to transform coordinates from one projection to another and use that to create WKT in the right projection immediately. Saves a parse/write WKT action. Also prevents the need to do a string-replace to create the CIRCLE WKT

Reply:
`ProjectionsHelper` was private, so I decided to instead create a `CoordinateHelper` which is exported, and currently just has a `projectCoordinates` function, which wraps `proj4` (this might be a bit unnecessary, but hey).